src/main/kotlin/org/hyacinthbots/lilybot/extensions/util/GuildAnnouncements.kt

```diff
@@ -0,0 +1,204 @@
+import com.kotlindiscord.kord.extensions.components.ephemeralButton
+import com.kotlindiscord.kord.extensions.extensions.Extension
+import com.kotlindiscord.kord.extensions.modules.unsafe.annotations.UnsafeAPI
+import com.kotlindiscord.kord.extensions.modules.unsafe.extensions.unsafeSlashCommand
+import com.kotlindiscord.kord.extensions.modules.unsafe.types.InitialSlashCommandResponse
+import com.kotlindiscord.kord.extensions.modules.unsafe.types.UnsafeInteractionContext
+import com.kotlindiscord.kord.extensions.utils.waitFor
+import dev.kord.common.Color
+import dev.kord.common.entity.ButtonStyle
+import dev.kord.common.entity.Permission
+import dev.kord.common.entity.Permissions
+import dev.kord.common.entity.TextInputStyle
+import dev.kord.core.behavior.UserBehavior
+import dev.kord.core.behavior.channel.createEmbed
+import dev.kord.core.behavior.interaction.modal
+import dev.kord.core.behavior.interaction.response.createEphemeralFollowup
+import dev.kord.core.behavior.interaction.response.edit
+import dev.kord.core.behavior.interaction.response.respond
+import dev.kord.core.entity.interaction.response.EphemeralMessageInteractionResponse
+import dev.kord.core.event.interaction.ModalSubmitInteractionCreateEvent
+import dev.kord.rest.builder.message.create.embed
+import kotlinx.coroutines.flow.toList
+import org.hyacinthbots.lilybot.utils.TEST_GUILD_ID
+import kotlin.time.Duration.Companion.seconds
+
+@OptIn(UnsafeAPI::class)
+class GuildAnnouncements : Extension() {
+	override val name = "guild-announcements"
+
+	override suspend fun setup() {
+		unsafeSlashCommand {
+			name = "announcement"
+			description = "Send an announcement to all guilds Lily is in"
+
+			initialResponse = InitialSlashCommandResponse.None
+
+			guild(TEST_GUILD_ID)
+
+			check {
+				hasPermission(Permission.Administrator)
+			}
+
+			action {
+				val modal = event.interaction.modal("Send an announcement", "announcementModal") {
+					actionRow {
+						textInput(TextInputStyle.Short, "title", "Announcement Title") {
+							placeholder = "Version 5.0.0!"
+						}
+					}
+
+					actionRow {
+						textInput(TextInputStyle.Paragraph, "body", "Announcement Body") {
+							placeholder = "This is a big update!"
+						}
+					}
+				}
+
+				val interaction =
+					modal.kord.waitFor<ModalSubmitInteractionCreateEvent>(300.seconds.inWholeMilliseconds) {
+						interaction.modalId == "announcementModal"
+					}?.interaction
+
+				if (interaction == null) {
+					modal.createEphemeralFollowup {
+						embed {
+							description = "Announcement timed out"
+						}
+					}
+					return@action
+				}
+
+				val title = interaction.textInputs["title"]!!.value!!
+				val body = interaction.textInputs["body"]!!.value!!
+				val footer = "Sent by ${user.asUser().tag}" // Useless, just needed for length calculations
+				val modalResponse = interaction.deferEphemeralResponse()
+
+				if (title.length + body.length + footer.length > 2048) {
```

Comment:
These should be evaluated seperately, currently it's possible to have ridiculously long titles. It might instead be worth simply having a content field, and have people make their own titles via bold.

Reply:
I see no problem with this though. So what if people want a long title.

Reply:
Also I have no clue what you mean by `evaluated seperately`. Outside the function or something??